### PR TITLE
Add Unit and Integration Tests

### PR DESCRIPTION
**Pull request by AlbertoMitroi**

### CODE REVIEW

File: src/Services/Discount/Discount.Grpc/discountdb-shm
Comment (ghost) [inline src/Services/Discount/Discount.Grpc/discountdb-shm:1]: Please add these files to `.gitignore`.

File: src/Services/Ordering/Ordering.Domain/ValueObjects/OrderName.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@ public record OrderName
         public static OrderName Of(string value)
         {
             ArgumentException.ThrowIfNullOrWhiteSpace(value);
-            ArgumentOutOfRangeException.ThrowIfNotEqual(value.Length, DefaultLength);
+            //ArgumentOutOfRangeException.ThrowIfNotEqual(value.Length, DefaultLength);
```
Comment (ghost) [inline src/Services/Ordering/Ordering.Domain/ValueObjects/OrderName.cs]: Is this still needed?

File: src/Services/Ordering/Ordering.Infrastructure/Data/Repositories/OrderRepository.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
+﻿using BuildingBlocks.Exceptions;
+
+namespace Ordering.Infrastructure.Data.Repositories
+{
+    public class OrderRepository(ApplicationDbContext _context) : IOrderRepository
+    {
+        public async Task<IEnumerable<Order>> GetAllAsync(CancellationToken cancellationToken)
+        {
+            IEnumerable<Order> orders = await _context.Orders.ToListAsync(cancellationToken);
```
Comment (ghost) [inline src/Services/Ordering/Ordering.Infrastructure/Data/Repositories/OrderRepository.cs]: Maybe consider add `.AsNoTracking()` as well to these if they are returned by the API?

Comment (ghost) [reply src/Services/Ordering/Ordering.Infrastructure/Data/Repositories/OrderRepository.cs]: Please review all methods below for this change.

File: src/Tests/Services/Ordering/Ordering.API.Application.Shared/BaseIntegrationTest.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace Ordering.Shared
+{
+    public abstract class BaseIntegrationTest : IClassFixture<IntegrationTestWebAppFactory>
+    {
+        private readonly IServiceScope _scope;
+        protected readonly ISender Sender;
+        protected readonly ApplicationDbContext _context;
+        private readonly IntegrationTestWebAppFactory factory;
+        protected readonly HttpClient _httpClient;
```
Comment (ghost) [inline src/Tests/Services/Ordering/Ordering.API.Application.Shared/BaseIntegrationTest.cs]: Please use `PascalCase` for `protected` members.

File: src/Tests/Services/Ordering/Ordering.API.Application.Shared/BaseIntegrationTest.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            Sender = _scope.ServiceProvider.GetRequiredService<ISender>();
+
+            _context = _scope.ServiceProvider.GetRequiredService<ApplicationDbContext>();
+
+            _httpClient = factory.CreateClient();
+
+            this.factory = factory;
+        }
+
+        protected IServiceScope CreateScope()
```
Comment (ghost) [inline src/Tests/Services/Ordering/Ordering.API.Application.Shared/BaseIntegrationTest.cs]: Is this still used?

File: src/Tests/Services/Ordering/Ordering.API.Application.Shared/IntegrationTestWebAppFactory.cs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+{
+    public class IntegrationTestWebAppFactory : WebApplicationFactory<Program>, IAsyncLifetime
+    {
+
+        private readonly MsSqlContainer _sqlContainer;
+        public IntegrationTestWebAppFactory()
+        {
+            _sqlContainer = new MsSqlBuilder()
+                .WithImage("mcr.microsoft.com/mssql/server:2022-CU15-GDR1-ubuntu-22.04")
+                .WithExposedPort("1435")
```
Comment (ghost) [inline src/Tests/Services/Ordering/Ordering.API.Application.Shared/IntegrationTestWebAppFactory.cs]: Is this `WithExposedPort` still needed?

File: src/Tests/Services/Ordering/Ordering.API.Application.Shared/IntegrationTestWebAppFactory.cs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                if (descriptor != null)
+                    services.Remove(descriptor);
+
+                services.AddDbContext<ApplicationDbContext>(options =>
+                {
+                    var connectionString = _sqlContainer.GetConnectionString();
+                    options.UseSqlServer(connectionString);
+                });
+
+                //var serviceProvider = services.BuildServiceProvider();
```
Comment (ghost) [inline src/Tests/Services/Ordering/Ordering.API.Application.Shared/IntegrationTestWebAppFactory.cs]: Please remove this commented code. You are already setting the database up in `Program.cs`.

File: src/Tests/Services/Ordering/Ordering.API.Application.Shared/IntegrationTestWebAppFactory.cs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                //    var db = scope.ServiceProvider.GetRequiredService<ApplicationDbContext>();
+                //    db.Database.EnsureCreated();
+                //}
+            });
+        }
+
+        public async Task InitializeAsync()
+        {
+            await _sqlContainer.StartAsync();
+        }
```
Comment (ghost) [inline src/Tests/Services/Ordering/Ordering.API.Application.Shared/IntegrationTestWebAppFactory.cs:45]: Add one empty line.  Please format your files using `CTRL+K,D`.

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/Services/Ordering/Ordering.Domain/ValueObjects/OrderName.cs (hunk 1 above)
Comment (AlbertoMitroi) [reply src/Services/Ordering/Ordering.Domain/ValueObjects/OrderName.cs]: no, i will remove it

File: src/Tests/Services/Ordering/Ordering.API.Application.Shared/IntegrationTestWebAppFactory.cs (hunk 5 above)
Comment (AlbertoMitroi) [reply src/Tests/Services/Ordering/Ordering.API.Application.Shared/IntegrationTestWebAppFactory.cs]: no. i will remove it

File: src/Services/Discount/Discount.Grpc/discountdb-shm
Comment (AlbertoMitroi) [reply src/Services/Discount/Discount.Grpc/discountdb-shm:1]: done

**Pull request merged**